persistence-api/src/main/java/io/stargate/db/datastore/PersistenceDataStoreFactory.java

```diff
@@ -69,8 +55,15 @@ private DataStore create(
       connection = persistence.newConnection();
     }
 
-    if (!isFromExternalAuth) {
-      connection.login(AuthenticatedUser.of(userName));
+    connection.login(user);
+
+    // Note: the user's custom properties (if provided) override any matching properties previously
+    // set in DataStoreOptions. This is intentional to give the AuthenticatedUser data more
+    // authority.
+    Map<String, String> customProperties = user.customProperties();
```

Comment:
Why do we need this override behavior?

Reply:
As explained in the comment - so that any properties provided by auth services through the `AuthenticationSubject` and `AuthenticatedUser` objects always take precedence over other values under the same key.